active/0003-extensible-resource-deployment-observability-improvements.md

```diff
@@ -0,0 +1,230 @@
+7. Create a new API version to expose the API changes and remove the feature flag.
+9. Update Swagger definition for the deployments and deployment operations APIs.
+10. Generate versions of Azure SDKs, including the Python SDK and C# SDK.
+11. Update client tools such as Azure CLI and Azure PowerShell to handle the new API responses.
+12. Collaborate with the Portal team to update the deployments blade, ensuring that the UI reflects the API changes.
+
+## Unresolved questions
+
+### Consistent schema V.S. Minimum backend changes
+
+To address the inconsistency in output resource schema between Azure resources and extensible resources, as highlighted in the Drawbacks section, a potential solution involves extending the schema to include additional properties for Azure resources as well, making Azure resources and extensible resources uniformly identifiable. For example:
+
+```diff
+{
+  ...
+  "type": "Microsoft.Resources/deployments",
+  "properties": {
+    "outputResources": [
+      // Azure resource
+      {
+        "id": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Storage/storageAccounts/my-storage-account"
++       "symbolicName": "myStorageAccount",
++       "resourceType": "Microsoft.Storage/storageAccounts",
++       "deploymentProvider": "az"
+      },
+      // Extensible resource
+      {
++       "symbolicName": "myService",
++       "resourceType": "core/Service@v1",
++       "deploymentProvider": "kubernetes"
+      }
+  }
+}
+```
+
+Implementing this enhancement involves comprehensive modifications across various deployment engine components, including but not limited to `AzureDeploymentEngine`, `ResourceProxyDefinition`, `DeploymentOutputResourceDefinition`, and numerous deployment jobs. Such widespread changes carry a significant risk due to the complexity of the deployment engine and the potential for unforeseen impacts on its functionality, which would require careful consideration and discussion.
+
+### Should the az provider always be returned?
+
+Building on the previous discussion, if the deploymentProvider details are not included within `outputResources` for Azure resources, should we consistently list the `az` provider in the `deploymentProviders` for sake of consistentcy and clarity?
+
+### Should `outputResources` include `existing` resources as well?
+
+Currently, `outputResources` only contains created or updated resources. Should we expand the property to include `existing` resources?
+
+## Out of scope
+
+### Portal UI design
+
+The update to the Portal deployments blade is necessary to display the symbolic name and deployment provider information for extensible resources. Since the portal team has ownership of the deployments blade, they hold the responsibility for both the UI design and the implementation of these changes. 
```

Comment:
I think the minimum we'd need is to make sure we give them a clear way to differentiate between an "Azure" and an extensible resource programmatically. I think you've covered that in this spec, but it might be worth stating it explicitly.

Reply:
Makes sense. Updated this section to mention it explicitly.